bluetooth: services: dis: add runtime configuration support (#812)
nrfconnect/sdk-nrf-bm: doc/nrf-bm/libraries/bluetooth/services/ble_dis.rst

During initialization, the module adds the Device Information Service to the Bluetooth LE stack database.
It then encodes the supplied information, and adds the corresponding characteristics.

> The characteristic values are taken either from run-time values supplied by the application or from the Kconfig options.

Reviewer: Might be formulated better than this, but I think it would be good to give some brief overview of when Kconfigs are used and not. 
```suggestion
The characteristic values can be configured during initialization of the service. If the values are not provided the defaults set by the Kconfig options are used.
```

Author: fixed but might need to edit again depending on the comments left by @lemrey and behaviour changes

Author: edited it to align with the API

* :kconfig:option:`CONFIG_BLE_DIS_MANUFACTURER_NAME` - Sets the manufacturer name.
* :kconfig:option:`CONFIG_BLE_DIS_MODEL_NUMBER` - Sets the model number.
* :kconfig:option:`CONFIG_BLE_DIS_SERIAL_NUMBER` - Sets the serial number.
* :kconfig:option:`CONFIG_BLE_DIS_HW_REVISION` - Sets the hardware revision.
* :kconfig:option:`CONFIG_BLE_DIS_FW_REVISION` - Sets the firmware revision.
* :kconfig:option:`CONFIG_BLE_DIS_SW_REVISION` - Sets the software revision.
* :kconfig:option:`CONFIG_BLE_DIS_SYSTEM_ID` - Includes the System ID characteristic in the Device Information Service.
* :kconfig:option:`CONFIG_BLE_DIS_SYSTEM_ID_OUI` - Sets the Organizationally Unique ID.
* :kconfig:option:`CONFIG_BLE_DIS_SYSTEM_ID_MID` - Sets the Manufacturer ID.
* :kconfig:option:`CONFIG_BLE_DIS_PNP_ID` - Includes the Plug and Play ID characteristic.
* :kconfig:option:`CONFIG_BLE_DIS_PNP_VID_SRC` - Sets the vendor ID source.
* :kconfig:option:`CONFIG_BLE_DIS_PNP_VID` - Sets the vendor ID.
* :kconfig:option:`CONFIG_BLE_DIS_PNP_PID` - Sets the product ID.
* :kconfig:option:`CONFIG_BLE_DIS_PNP_VER` - Sets the product version.

> * :kconfig:option:`CONFIG_BLE_DIS_REGULATORY_CERT` - Includes the IEEE 11073-20601 Regulatory Certification characteristic.

Reviewer: This does not belong in this PR, at least not in the commit introducing the runtime configuration.

Author: fixed, made own commit for this

During initialization, the module adds the Device Information Service to the Bluetooth LE stack database.
It then encodes the supplied information, and adds the corresponding characteristics.

The characteristic values follow an all-or-nothing model:

> * If the application supplies a :c:struct:`ble_dis_values` instance at initialization, those values fully define every characteristic.

Reviewer 2: ```suggestion
* If the application supplies a :c:struct:`ble_dis_values` instance at initialization, the values fully define every characteristic.
```

Author: fixed

>   The Kconfig options are ignored, and any member left ``NULL`` (or empty for strings) omits that characteristic.

Reviewer 2: ```suggestion
  The Kconfig options are ignored and any member left ``NULL`` (or empty for strings) omits that characteristic.
```

Author: fixed

* If no run-time values are supplied, all characteristics are built from the :kconfig:option:`CONFIG_BLE_DIS_*` Kconfig build-time defaults.

Configuration
*************

Set the :kconfig:option:`CONFIG_BLE_DIS` Kconfig option to enable the service.

> The service can be configured using the following Kconfig options, which provide the build-time default values:

Reviewer 2: ```suggestion
You can configure the service using the following Kconfig options that provide the build-time default values:
```

Author: fixed

* :kconfig:option:`CONFIG_BLE_DIS_HW_REVISION` - Sets the hardware revision.
* :kconfig:option:`CONFIG_BLE_DIS_FW_REVISION` - Sets the firmware revision.
* :kconfig:option:`CONFIG_BLE_DIS_SW_REVISION` - Sets the software revision.
* :kconfig:option:`CONFIG_BLE_DIS_SYSTEM_ID` - Includes the System ID characteristic in the Device Information Service.
* :kconfig:option:`CONFIG_BLE_DIS_SYSTEM_ID_OUI` - Sets the Organizationally Unique ID.
* :kconfig:option:`CONFIG_BLE_DIS_SYSTEM_ID_MID` - Sets the Manufacturer ID.
* :kconfig:option:`CONFIG_BLE_DIS_PNP_ID` - Includes the Plug and Play ID characteristic.
* :kconfig:option:`CONFIG_BLE_DIS_PNP_VID_SRC` - Sets the vendor ID source.
* :kconfig:option:`CONFIG_BLE_DIS_PNP_VID` - Sets the vendor ID.
* :kconfig:option:`CONFIG_BLE_DIS_PNP_PID` - Sets the product ID.
* :kconfig:option:`CONFIG_BLE_DIS_PNP_VER` - Sets the product version.
* :kconfig:option:`CONFIG_BLE_DIS_REGULATORY_CERT` - Includes the IEEE 11073-20601 Regulatory Certification characteristic.
* :kconfig:option:`CONFIG_BLE_DIS_REGULATORY_CERT_LIST` - Sets the regulatory certification list.

Initialization
==============

> The service is initialized by calling the :c:func:`ble_dis_init` function.

Reviewer 2: ```suggestion
To initialize the service, call the :c:func:`ble_dis_init` function.
```

Author: fixed
